### PR TITLE
Call super.config() in BartBackbone's get_config()

### LINKED ISSUE #813: Call `super.config()` in `BartBackbone`'s `get_config()`
We should call `super().config()` here: https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/models/bart/bart_backbone.py#L238, update the config with `BARTBackbone`-specific keys and return the config. Check this for reference: https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/models/bert/bert_backbone.py#L204.

### PR DESCRIPTION
**Pull request by shivance**
Closes #813

### CODE REVIEW

File: keras_nlp/models/bart/bart_backbone.py [hunk 1]
```diff
@@ -236,17 +236,20 @@ def __init__(
...
+                "num_layers": self.num_layers,
+                "num_heads": self.num_heads,
+                "hidden_dim": self.hidden_dim,
+                "intermediate_dim": self.intermediate_dim,
+                "dropout": self.dropout,
+                "max_sequence_length": self.max_sequence_length,
+                "name": self.name,
+                "trainable": self.trainable,
+            }
+        )
```
Comment (mattdangerw) [inline keras_nlp/models/bart/bart_backbone.py:252]: missing return! causing some test failures

**Review by mattdangerw — CHANGES_REQUESTED** [review]

File: keras_nlp/models/bart/bart_backbone.py [hunk 2]
```diff
@@ -236,17 +236,22 @@ def __init__(
...
+        config.update(
+            {
+                "vocabulary_size": self.vocabulary_size,
+                "num_layers": self.num_layers,
+                "num_heads": self.num_heads,
+                "hidden_dim": self.hidden_dim,
+                "intermediate_dim": self.intermediate_dim,
+                "dropout": self.dropout,
+                "max_sequence_length": self.max_sequence_length,
+                "name": self.name,
```
Comment (mattdangerw) [inline keras_nlp/models/bart/bart_backbone.py:249]: when chaining to super we should now be able to remove `name` and `trainable` here (they are handled on the super class)

**Review by mattdangerw — APPROVED** [review]
one last comment, then let's merge

File: keras_nlp/models/bart/bart_backbone.py (hunk 2 above)
Comment (shivance) [reply keras_nlp/models/bart/bart_backbone.py]: Interesting !  Done.

**Comment by mattdangerw** [comment]
LGTM!

**Comment by shivance** [comment]
Thanks @mattdangerw ! Just a gentle request that #754 and #794 need review !

**Pull request merged**